RCNN_tf.py

```diff
@@ -68,46 +93,81 @@ def rcl(
         name=name + '_bias'
     )
 
-    forward_output = tf.nn.conv2d(input_data, forward_weights, [1, 1, 1, 1], padding='SAME')
+    forward_output = tf.nn.conv2d(
+        input_data,
+        forward_weights,
+        [1, 1, 1, 1],
+        padding='SAME'
+    )
+
+    if "CPU" in processing_unit.upper():
+        def termination_cond():
+            """Specifies the termination condition for the Tensorflow
+            while loop"""
+            return lambda x, recurrent_states, output: x < depth
+
 
-    if "CPU" in device_name.upper():
         def loop_body(x, recurrent_states, output):
-            recurrent_output = tf.nn.conv2d(recurrent_states, recurrent_weights, [1, 1, 1, 1], padding='SAME')
+            """Specifies the body callable of the Tensorflow while loop
+
+            Args/Returns:
+                x                (tf.Tensor): counter
+                recurrent_states (tf.Tensor): the recurrent states of the
+                                              recurrent convolutional layer
+                output           (tf.Tensor): the output tensor, which is the
+                                              addition of the forward and
+                                              recurrent output and the bias
+            """
+            recurrent_output = tf.nn.conv2d(
+                    recurrent_states,
+                    recurrent_weights,
+                    [1, 1, 1, 1],
+                    padding='SAME')
 
             output = tf.add(forward_output, recurrent_output)
             output += bias
 
             recurrent_states = tf.nn.local_response_normalization(
                 tf.maximum(np.asarray(0.0).astype(PRECISION_NP), output),
-                depth_radius=normalization_feature_maps,
+                depth_radius=num_norm_feat_maps,
                 bias=1,
                 alpha=alpha,
                 beta=beta,
                 name=name + '_lrn'
             )
+
             x += 1
+
             return x, recurrent_states, output
 
         results = tf.while_loop(
-            lambda x, recurrent_states, output: x < depth,
-            loop_body,
-            [0, cell_states, output],
+            cond=termination_cond(),
```

Comment:
I don't think this works. `x`, `recurrent_states`, and `output` should be passed as parameter.
I will change that

Reply:
I tested it and it seems to work just fine -- in my opinion it makes the code a bit easier to understand.